### PR TITLE
curve factory deposit script, curvev2, univ2, sushi classes

### PR DESCRIPTION
**Pull request by jalbrekt85**
- script addressing #34 to deposit into equal parts badger/wbtc into curve factory pool and monitors pool data. let me know if I have the right idea
- `greatapesafe` classes for curveV2, uniV2, and sushi
- ~~no withdraw functionality for curveV2 yet~~
- simple curvev2 class
- curve class updated to work with v2 interfaces

### CODE REVIEW

File: great_ape_safe/ape_api/curve.py [hunk 1]
```diff
@@ -31,56 +32,75 @@ def deposit(self, lp_token, mantissas, asset=None):
         # TODO: find and route through zaps automatically
         # TODO: could pass dict to mantissas with {address: mantissa} and sort
         #       out proper ordering automatically?
-        pool_addr = self.registry.get_pool_from_lp_token(lp_token)
-        pool = self.safe.contract(pool_addr)
+        
+        # pool must be provided; cannot look up v2 factory pools from lp token
```
Comment (jorijnsmit) [inline great_ape_safe/ape_api/curve.py:36]: looks like `lp_token.minter()` should do the trick:
```
>>> from helpers.addresses import registry
>>> lp_token = interface.ICurveLP(registry.eth.treasury_tokens.badgerWBTC_f)
>>> lp_token.minter()
'0x50f3752289e1456BfA505afd37B241bca23e685d'
>>> registry.eth.crv_factory_pools.badgerWBTC_f
'0x50f3752289e1456BfA505afd37B241bca23e685d'
```

**Review by jorijnsmit — CHANGES_REQUESTED** [review]
good use of the interfaces!!!

minor comments added inline.

File: great_ape_safe/ape_api/curve.py (hunk 1 above)
Comment (jalbrekt85) [reply great_ape_safe/ape_api/curve.py:36]: forgot that method existed, very helpful

File: scripts/issue/34/issue34.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    safe.init_curve_v2()
+
+    if 'fork' in network.show_active():
+        WBTC = MintableForkToken(registry.eth.treasury_tokens.WBTC)
+        BADGER = MintableForkToken(registry.eth.treasury_tokens.BADGER)
+        WBTC._mint_for_testing(safe.address, 3.5 * 10**WBTC.decimals())
+        BADGER._mint_for_testing(safe.address, 15000 * 10**BADGER.decimals())
+    else:
+        WBTC = interface.ERC20(registry.eth.treasury_tokens.WBTC)
+        BADGER = interface.ERC20(registry.eth.treasury_tokens.BADGER)
```
Comment (jorijnsmit) [inline scripts/issue/34/issue34.py:35]: this will need some adjusting

**Review by jorijnsmit — APPROVED** [review]
looking good @jalbrekt85 ! the issue34.py file itself we will have to adjust a bit (we always post from fork for example, so L28-L35 isnt going to work) but lets merge this since it provides lots of helpful functions already!

**Comment by jorijnsmit** [comment]
@petrovska-petro your old code here being merged into greatape :)

**Review by petrovska-petro — APPROVED** [review]
Lots of goodies added into this PR!

Thanks for porting some of the methods of the old repo into here 🔥 

Talked with Gos, that probably the issue script itself will need some changes, rather than adding in one whole lump sum

**Pull request merged**